cpp/ql/src/semmle/code/cpp/security/FlowSources.qll

```diff
@@ -0,0 +1,35 @@
+/**
+ * Provides classes representing various flow sources for taint tracking.
+ */
+
+import cpp
+import semmle.code.cpp.ir.dataflow.DataFlow
+private import semmle.code.cpp.ir.IR
+import semmle.code.cpp.models.interfaces.FlowSource
+
+/** A data flow source of remote user input. */
+abstract class RemoteFlowSource extends DataFlow::Node {
+}
+
+class FileDescriptorTaintedReturnSource extends RemoteFlowSource {
+  FileDescriptorTaintedReturnSource() {
```

Comment:
```suggestion
private class TaintedReturnSource extends RemoteFlowSource {
  TaintedReturnSource() {
```

This is not restricted to just file descriptors, right?

Reply:
It's not.